### ISSUE TITLE
[clang] clang-linker-wrapper crashes with -g option 

### ISSUE THREAD

**Issue by fel-cab** [llvm/llvm-project#85472]
OpenMP offloading applications on AMD GPU system crashes when compiled with `-g` flag in assert: 
from `llvm/lib/IR/Instruction.cpp:163 `

`assert(!isa<PHINode>(this) && "Inserting PHI after debug-records!");`

Could not make a small reproducible 

First Crash:
```
clang-linker-wrapper: /lustre/orion/ums012/scratch/cabarcasf/ums012/llvm17_git/llvm-project/llvm/lib/IR/Instruction.cpp:163: void llvm::Instruction::insertBefore(BasicBlock &, InstListType::iterator): Assertion `!isa<PHINode>(this) && "Inserting PHI after debug-records!"' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper --opt-level=O3 --host-triple=x86_64-unknown-linux-gnu --device-debug --linker-path=/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld --hash-style=gnu --eh-frame-hdr -m elf_x86_64 -pie -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o hpgmgfv /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/Scrt1.o /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o /usr/lib64/gcc/x86_64-suse-linux/7/crtbeginS.o /usr/lib64/gcc/x86_64-suse-linux/7/crtfastmath.o -L/opt/cray/pe/mpich/8.1.23/ofi/cray/10.0/lib -L/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/lib/clang/19/lib/x86_64-unknown-linux-gnu -L/usr/lib64/gcc/x86_64-suse-linux/7 -L/usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/lib -L/lib -L/usr/lib -lomptarget -lm hpgmg-fv.o level.o mg.o operators.fv4.o solvers.o timers.o directives-wrapper.o cuda-wrapper.o offload-fns.o specmpitime/specmpitime.o -lm -lmpi_cray -lomp -lomptarget -lomptarget.devicertl -L/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/lib -lgcc --as-needed -lgcc_s --no-as-needed -lpthread -lc -lgcc --as-needed -lgcc_s --no-as-needed /usr/lib64/gcc/x86_64-suse-linux/7/crtendS.o /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crtn.o
1.	Running pass 'Function Pass Manager' on module 'ld-temp.o'.
2.	Running pass 'AMDGPU IR optimizations' on function '@__omp_offloading_8116438_430033dc_directives_apply_BCs_v2_l161'
 #0 0x000056152fd06718 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x12d6718)
 #1 0x000056152fd0439e llvm::sys::RunSignalHandlers() (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x12d439e)
 #2 0x000056152fd06f2d SignalHandler(int) Signals.cpp:0:0
 #3 0x00001491fad8a8c0 __restore_rt (/lib64/libpthread.so.0+0x168c0)
 #4 0x00001491f97eacbb raise (/lib64/libc.so.6+0x4acbb)
 #5 0x00001491f97ec355 abort (/lib64/libc.so.6+0x4c355)
 #6 0x00001491f97e2cba __assert_fail_base (/lib64/libc.so.6+0x42cba)
 #7 0x00001491f97e2d42 (/lib64/libc.so.6+0x42d42)
 #8 0x000056152f6d8970 llvm::Instruction::insertAfter(llvm::Instruction*) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca8970)
 #9 0x000056152f6d83d7 llvm::Instruction::insertInto(llvm::BasicBlock*, llvm::ilist_iterator_w_bits<llvm::ilist_detail::node_options<llvm::Instruction, true, false, void, true>, false, false>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca83d7)
#10 0x000056152f6d2760 llvm::IRBuilderDefaultInserter::InsertHelper(llvm::Instruction*, llvm::Twine const&, llvm::BasicBlock*, llvm::ilist_iterator_w_bits<llvm::ilist_detail::node_options<llvm::Instruction, true, false, void, true>, false, false>) const (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca2760)
#11 0x000056152eda1582 llvm::IRBuilderBase::CreatePHI(llvm::Type*, unsigned int, llvm::Twine const&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x371582)
#12 0x000056152f2e5c87 (anonymous namespace)::AMDGPUCodeGenPrepareImpl::run(llvm::Function&) AMDGPUCodeGenPrepare.cpp:0:0
#13 0x000056152f2f02d5 (anonymous namespace)::AMDGPUCodeGenPrepare::runOnFunction(llvm::Function&) AMDGPUCodeGenPrepare.cpp:0:0
#14 0x000056152f70d707 llvm::FPPassManager::runOnFunction(llvm::Function&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xcdd707)
#15 0x000056152f715bf2 llvm::FPPassManager::runOnModule(llvm::Module&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xce5bf2)
#16 0x000056152f70e1ff llvm::legacy::PassManagerImpl::run(llvm::Module&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xcde1ff)
#17 0x000056153023cac6 codegen(llvm::lto::Config const&, llvm::TargetMachine*, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex const&) LTOBackend.cpp:0:0
#18 0x000056153023ba72 llvm::lto::backend(llvm::lto::Config const&, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x180ba72)
#19 0x000056153022dd93 llvm::lto::LTO::runRegularLTO(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x17fdd93)
#20 0x000056153022d377 llvm::lto::LTO::run(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, std::function<llvm::Expected<std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>> (unsigned int, llvm::StringRef, llvm::Twine const&)>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x17fd377)
#21 0x000056152ed18f49 (anonymous namespace)::linkBitcodeFiles(llvm::SmallVectorImpl<llvm::object::OffloadFile>&, llvm::SmallVectorImpl<llvm::StringRef>&, llvm::opt::ArgList const&) ClangLinkerWrapper.cpp:0:0
#22 0x000056152ed122fe _ZZN12_GLOBAL__N_122linkAndWrapDeviceFilesERN4llvm15SmallVectorImplINS0_11SmallVectorINS0_6object11OffloadFileELj3EEEEERKNS0_3opt12InputArgListEPPciENK3$_0clIS5_EENS0_5ErrorERT_ ClangLinkerWrapper.cpp:0:0
#23 0x000056152ed08d5d (anonymous namespace)::linkAndWrapDeviceFiles(llvm::SmallVectorImpl<llvm::SmallVector<llvm::object::OffloadFile, 3u>>&, llvm::opt::InputArgList const&, char**, int) ClangLinkerWrapper.cpp:0:0
#24 0x000056152ed01ee3 main (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x2d1ee3)
#25 0x00001491f97d529d __libc_start_main (/lib64/libc.so.6+0x3529d)
#26 0x000056152ed009ba _start /home/abuild/rpmbuild/BUILD/glibc-2.31/csu/../sysdeps/x86_64/start.S:122:0
 #0 0x000056152fd06718 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x12d6718)
 #1 0x000056152fd043d5 llvm::sys::RunSignalHandlers() (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x12d43d5)
 #2 0x000056152fd06f2d SignalHandler(int) Signals.cpp:0:0
 #3 0x00001491fad8a8c0 __restore_rt (/lib64/libpthread.so.0+0x168c0)
 #4 0x00001491f97eacbb raise (/lib64/libc.so.6+0x4acbb)
 #5 0x00001491f97ec355 abort (/lib64/libc.so.6+0x4c355)
 #6 0x00001491f97e2cba __assert_fail_base (/lib64/libc.so.6+0x42cba)
 #7 0x00001491f97e2d42 (/lib64/libc.so.6+0x42d42)
 #8 0x000056152f6d8970 llvm::Instruction::insertAfter(llvm::Instruction*) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca8970)
 #9 0x000056152f6d83d7 llvm::Instruction::insertInto(llvm::BasicBlock*, llvm::ilist_iterator_w_bits<llvm::ilist_detail::node_options<llvm::Instruction, true, false, void, true>, false, false>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca83d7)
#10 0x000056152f6d2760 llvm::IRBuilderDefaultInserter::InsertHelper(llvm::Instruction*, llvm::Twine const&, llvm::BasicBlock*, llvm::ilist_iterator_w_bits<llvm::ilist_detail::node_options<llvm::Instruction, true, false, void, true>, false, false>) const (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca2760)
#11 0x000056152eda1582 llvm::IRBuilderBase::CreatePHI(llvm::Type*, unsigned int, llvm::Twine const&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x371582)
#12 0x000056152f2e5c87 (anonymous namespace)::AMDGPUCodeGenPrepareImpl::run(llvm::Function&) AMDGPUCodeGenPrepare.cpp:0:0
#13 0x000056152f2f02d5 (anonymous namespace)::AMDGPUCodeGenPrepare::runOnFunction(llvm::Function&) AMDGPUCodeGenPrepare.cpp:0:0
#14 0x000056152f70d707 llvm::FPPassManager::runOnFunction(llvm::Function&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xcdd707)
#15 0x000056152f715bf2 llvm::FPPassManager::runOnModule(llvm::Module&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xce5bf2)
#16 0x000056152f70e1ff llvm::legacy::PassManagerImpl::run(llvm::Module&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xcde1ff)
#17 0x000056153023cac6 codegen(llvm::lto::Config const&, llvm::TargetMachine*, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex const&) LTOBackend.cpp:0:0
#18 0x000056153023ba72 llvm::lto::backend(llvm::lto::Config const&, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x180ba72)
#19 0x000056153022dd93 llvm::lto::LTO::runRegularLTO(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x17fdd93)
#20 0x000056153022d377 llvm::lto::LTO::run(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, std::function<llvm::Expected<std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>> (unsigned int, llvm::StringRef, llvm::Twine const&)>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x17fd377)
#21 0x000056152ed18f49 (anonymous namespace)::linkBitcodeFiles(llvm::SmallVectorImpl<llvm::object::OffloadFile>&, llvm::SmallVectorImpl<llvm::StringRef>&, llvm::opt::ArgList const&) ClangLinkerWrapper.cpp:0:0
#22 0x000056152ed122fe _ZZN12_GLOBAL__N_122linkAndWrapDeviceFilesERN4llvm15SmallVectorImplINS0_11SmallVectorINS0_6object11OffloadFileELj3EEEEERKNS0_3opt12InputArgListEPPciENK3$_0clIS5_EENS0_5ErrorERT_ ClangLinkerWrapper.cpp:0:0
#23 0x000056152ed08d5d (anonymous namespace)::linkAndWrapDeviceFiles(llvm::SmallVectorImpl<llvm::SmallVector<llvm::object::OffloadFile, 3u>>&, llvm::opt::InputArgList const&, char**, int) ClangLinkerWrapper.cpp:0:0
#24 0x000056152ed01ee3 main (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x2d1ee3)
#25 0x00001491f97d529d __libc_start_main (/lib64/libc.so.6+0x3529d)
#26 0x000056152ed009ba _start /home/abuild/rpmbuild/BUILD/glibc-2.31/csu/../sysdeps/x86_64/start.S:122:0
clang: error: unable to execute command: Aborted (core dumped)
clang: error: linker command failed due to signal (use -v to see invocation)
```

Second Crash:
```
clang-linker-wrapper: /lustre/orion/ums012/scratch/cabarcasf/ums012/llvm17_git/llvm-project/llvm/lib/IR/Instruction.cpp:163: void llvm::Instruction::insertBefore(BasicBlock &, InstListType::iterator): Assertion `!isa<PHINode>(this) && "Inserting PHI after debug-records!"' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper --opt-level=O3 --host-triple=x86_64-unknown-linux-gnu --device-debug --linker-path=/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld --hash-style=gnu --eh-frame-hdr -m elf_x86_64 -pie -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o soma /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/Scrt1.o /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o /usr/lib64/gcc/x86_64-suse-linux/7/crtbeginS.o /usr/lib64/gcc/x86_64-suse-linux/7/crtfastmath.o -L/opt/cray/pe/mpich/8.1.23/ofi/cray/10.0/lib -L/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/lib/clang/19/lib/x86_64-unknown-linux-gnu -L/usr/lib64/gcc/x86_64-suse-linux/7 -L/usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/lib -L/lib -L/usr/lib -lomptarget -lm allocator.o ana.o autotuner.o bond.o cmdline.o device.o generate_positions.o independent_sets.o init.o io.o mc.o mesh.o monomer.o mpiroutines.o phase.o polymer.o rng.o soma.o soma_config.o soma_util.o test.o specmpitime/specmpitime.o -lm -lmpi_cray -lomp -lomptarget -lomptarget.devicertl -L/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/lib -lgcc --as-needed -lgcc_s --no-as-needed -lpthread -lc -lgcc --as-needed -lgcc_s --no-as-needed /usr/lib64/gcc/x86_64-suse-linux/7/crtendS.o /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crtn.o
1.	Running pass 'Function Pass Manager' on module 'ld-temp.o'.
2.	Running pass 'AMDGPU IR optimizations' on function '@__omp_offloading_8116438_43003429_generate_new_beads_l184'
 #0 0x000055ce02f33718 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x12d6718)
 #1 0x000055ce02f3139e llvm::sys::RunSignalHandlers() (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x12d439e)
 #2 0x000055ce02f33f2d SignalHandler(int) Signals.cpp:0:0
 #3 0x000014dab332e8c0 __restore_rt (/lib64/libpthread.so.0+0x168c0)
 #4 0x000014dab1d8ecbb raise (/lib64/libc.so.6+0x4acbb)
 #5 0x000014dab1d90355 abort (/lib64/libc.so.6+0x4c355)
 #6 0x000014dab1d86cba __assert_fail_base (/lib64/libc.so.6+0x42cba)
 #7 0x000014dab1d86d42 (/lib64/libc.so.6+0x42d42)
 #8 0x000055ce02905970 llvm::Instruction::insertAfter(llvm::Instruction*) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca8970)
 #9 0x000055ce029053d7 llvm::Instruction::insertInto(llvm::BasicBlock*, llvm::ilist_iterator_w_bits<llvm::ilist_detail::node_options<llvm::Instruction, true, false, void, true>, false, false>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca83d7)
#10 0x000055ce028ff760 llvm::IRBuilderDefaultInserter::InsertHelper(llvm::Instruction*, llvm::Twine const&, llvm::BasicBlock*, llvm::ilist_iterator_w_bits<llvm::ilist_detail::node_options<llvm::Instruction, true, false, void, true>, false, false>) const (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca2760)
#11 0x000055ce01fce582 llvm::IRBuilderBase::CreatePHI(llvm::Type*, unsigned int, llvm::Twine const&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x371582)
#12 0x000055ce02512c87 (anonymous namespace)::AMDGPUCodeGenPrepareImpl::run(llvm::Function&) AMDGPUCodeGenPrepare.cpp:0:0
#13 0x000055ce0251d2d5 (anonymous namespace)::AMDGPUCodeGenPrepare::runOnFunction(llvm::Function&) AMDGPUCodeGenPrepare.cpp:0:0
#14 0x000055ce0293a707 llvm::FPPassManager::runOnFunction(llvm::Function&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xcdd707)
#15 0x000055ce02942bf2 llvm::FPPassManager::runOnModule(llvm::Module&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xce5bf2)
#16 0x000055ce0293b1ff llvm::legacy::PassManagerImpl::run(llvm::Module&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xcde1ff)
#17 0x000055ce03469ac6 codegen(llvm::lto::Config const&, llvm::TargetMachine*, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex const&) LTOBackend.cpp:0:0
#18 0x000055ce03468a72 llvm::lto::backend(llvm::lto::Config const&, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x180ba72)
#19 0x000055ce0345ad93 llvm::lto::LTO::runRegularLTO(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x17fdd93)
#20 0x000055ce0345a377 llvm::lto::LTO::run(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, std::function<llvm::Expected<std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>> (unsigned int, llvm::StringRef, llvm::Twine const&)>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x17fd377)
#21 0x000055ce01f45f49 (anonymous namespace)::linkBitcodeFiles(llvm::SmallVectorImpl<llvm::object::OffloadFile>&, llvm::SmallVectorImpl<llvm::StringRef>&, llvm::opt::ArgList const&) ClangLinkerWrapper.cpp:0:0
#22 0x000055ce01f3f2fe _ZZN12_GLOBAL__N_122linkAndWrapDeviceFilesERN4llvm15SmallVectorImplINS0_11SmallVectorINS0_6object11OffloadFileELj3EEEEERKNS0_3opt12InputArgListEPPciENK3$_0clIS5_EENS0_5ErrorERT_ ClangLinkerWrapper.cpp:0:0
#23 0x000055ce01f35d5d (anonymous namespace)::linkAndWrapDeviceFiles(llvm::SmallVectorImpl<llvm::SmallVector<llvm::object::OffloadFile, 3u>>&, llvm::opt::InputArgList const&, char**, int) ClangLinkerWrapper.cpp:0:0
#24 0x000055ce01f2eee3 main (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x2d1ee3)
#25 0x000014dab1d7929d __libc_start_main (/lib64/libc.so.6+0x3529d)
#26 0x000055ce01f2d9ba _start /home/abuild/rpmbuild/BUILD/glibc-2.31/csu/../sysdeps/x86_64/start.S:122:0
 #0 0x000055ce02f33718 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x12d6718)
 #1 0x000055ce02f313d5 llvm::sys::RunSignalHandlers() (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x12d43d5)
 #2 0x000055ce02f33f2d SignalHandler(int) Signals.cpp:0:0
 #3 0x000014dab332e8c0 __restore_rt (/lib64/libpthread.so.0+0x168c0)
 #4 0x000014dab1d8ecbb raise (/lib64/libc.so.6+0x4acbb)
 #5 0x000014dab1d90355 abort (/lib64/libc.so.6+0x4c355)
 #6 0x000014dab1d86cba __assert_fail_base (/lib64/libc.so.6+0x42cba)
 #7 0x000014dab1d86d42 (/lib64/libc.so.6+0x42d42)
 #8 0x000055ce02905970 llvm::Instruction::insertAfter(llvm::Instruction*) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca8970)
 #9 0x000055ce029053d7 llvm::Instruction::insertInto(llvm::BasicBlock*, llvm::ilist_iterator_w_bits<llvm::ilist_detail::node_options<llvm::Instruction, true, false, void, true>, false, false>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca83d7)
#10 0x000055ce028ff760 llvm::IRBuilderDefaultInserter::InsertHelper(llvm::Instruction*, llvm::Twine const&, llvm::BasicBlock*, llvm::ilist_iterator_w_bits<llvm::ilist_detail::node_options<llvm::Instruction, true, false, void, true>, false, false>) const (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xca2760)
#11 0x000055ce01fce582 llvm::IRBuilderBase::CreatePHI(llvm::Type*, unsigned int, llvm::Twine const&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x371582)
#12 0x000055ce02512c87 (anonymous namespace)::AMDGPUCodeGenPrepareImpl::run(llvm::Function&) AMDGPUCodeGenPrepare.cpp:0:0
#13 0x000055ce0251d2d5 (anonymous namespace)::AMDGPUCodeGenPrepare::runOnFunction(llvm::Function&) AMDGPUCodeGenPrepare.cpp:0:0
#14 0x000055ce0293a707 llvm::FPPassManager::runOnFunction(llvm::Function&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xcdd707)
#15 0x000055ce02942bf2 llvm::FPPassManager::runOnModule(llvm::Module&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xce5bf2)
#16 0x000055ce0293b1ff llvm::legacy::PassManagerImpl::run(llvm::Module&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0xcde1ff)
#17 0x000055ce03469ac6 codegen(llvm::lto::Config const&, llvm::TargetMachine*, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex const&) LTOBackend.cpp:0:0
#18 0x000055ce03468a72 llvm::lto::backend(llvm::lto::Config const&, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex&) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x180ba72)
#19 0x000055ce0345ad93 llvm::lto::LTO::runRegularLTO(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x17fdd93)
#20 0x000055ce0345a377 llvm::lto::LTO::run(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>, std::function<llvm::Expected<std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream>>> (unsigned int, llvm::Twine const&)>> (unsigned int, llvm::StringRef, llvm::Twine const&)>) (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x17fd377)
#21 0x000055ce01f45f49 (anonymous namespace)::linkBitcodeFiles(llvm::SmallVectorImpl<llvm::object::OffloadFile>&, llvm::SmallVectorImpl<llvm::StringRef>&, llvm::opt::ArgList const&) ClangLinkerWrapper.cpp:0:0
#22 0x000055ce01f3f2fe _ZZN12_GLOBAL__N_122linkAndWrapDeviceFilesERN4llvm15SmallVectorImplINS0_11SmallVectorINS0_6object11OffloadFileELj3EEEEERKNS0_3opt12InputArgListEPPciENK3$_0clIS5_EENS0_5ErrorERT_ ClangLinkerWrapper.cpp:0:0
#23 0x000055ce01f35d5d (anonymous namespace)::linkAndWrapDeviceFiles(llvm::SmallVectorImpl<llvm::SmallVector<llvm::object::OffloadFile, 3u>>&, llvm::opt::InputArgList const&, char**, int) ClangLinkerWrapper.cpp:0:0
#24 0x000055ce01f2eee3 main (/autofs/nccs-svm1_sw/frontier/ums/ums012/llvm/19.0.0-20240315/bin/clang-linker-wrapper+0x2d1ee3)
#25 0x000014dab1d7929d __libc_start_main (/lib64/libc.so.6+0x3529d)
#26 0x000055ce01f2d9ba _start /home/abuild/rpmbuild/BUILD/glibc-2.31/csu/../sysdeps/x86_64/start.S:122:0
clang: error: unable to execute command: Aborted (core dumped)
clang: error: linker command failed due to signal (use -v to see invocation)
```

**Comment by OCHyams** [comment]
Thanks for the report. Looking at the stack trace I think this should fix it - #85618.

**Comment by fel-cab** [comment]
Thanks for the fix, working now.
